### ISSUE TITLE
🧩 Increasing the flexibility of writing plugins by providing the already_in_the_DOM var in the request

### ISSUE THREAD

**Issue by Ingenu1ty** [saltcorn/saltcorn#2492]
Formulating this FR was pretty challenging and the google translate made it even more weird :-), so I made a video to explain what I want to build: 
[dlist_demo.webm](https://github.com/saltcorn/saltcorn/assets/138471707/38027bb3-895e-4913-903d-39c609480175)

a custom list plugin that receives, along with the request, a variable indicating that this **view has already been rendered** with a state: "some_state_". And in this case it should be able to return **not static html**, but something like this: { eval_js: some_js_that_will_manipulate_the_DOM } 

in other words: 

1)case: 
the page loads for the first time - Everything happens as usual. 

2)case: 
The state on an **already loaded** page changes(For example, the filter button is pressed). - return something like this: { eval_js: some_js_that_will_manipulate_the_DOM }

P.S. These changes are not just idle decoration, but an [important thing in UX](https://www.geeksforgeeks.org/the-role-of-motion-design-in-ux/#:~:text=does%20motion%20design-,Role%20of%20Motion%20design%20in%20UX,-There%20are%20five).

**Comment by glutamate** [comment]
I think we should look at the new view transitions API for this.

**Comment by Ingenu1ty** [comment]
I agree. A clear and simple API would be a better architectural solution. But I assume that its implementation will be much more complicated... How, while maintaining server side rendering, provide the necessary flexibility for effects on the browser side?(please note that in my video example, not just one html chunk replaces another, but each list_item behaves differently, depending on the state - This way, motion design shows the user **which items have been added, which have gone, and which remain unchanged**) 

I think we can look towards mechanics similar to [hydration](https://themobilereality.com/blog/hydration-ssr-with-react-18#:~:text=The%20beauty%20of,and%20SEO%20rankings.) or something like [that...](https://remix.run/docs/en/main/discussion/introduction#:~:text=But%20as%20soon,default%20browser%20behaviors.)
